subliminal/providers/subscene.py

```diff
@@ -0,0 +1,334 @@
+            if "Year" in li.strong.string:
+                year = int(li.strong.next_sibling.strip())
+                break
+
+        # if just one language is requested, SSF is available and all the
+        # results will be of specified language; no need to parse lang field.
+        single_lang = langs.pop() if len(langs) == 1 else None
+
+        for tr in soup.table.tbody.children:
+            if tr.name != "tr":
+                continue
+            a = tr.find("td", class_="a1")
+            if a is None:
+                continue
+            a = a.a
+            if single_lang is not None:
+                lang = single_lang
+            else:
+                lang = a.find("span").string.strip()
+                try:
+                    lang = Language.fromsubscene(lang)
+                except NotImplementedError:
+                    continue
+                if manual_filter and lang not in langs:
+                    continue
+
+            page_link = a["href"]
+            desc = a.contents[1].string.strip()
+            num_files = tr.find("td", class_="a3").string.strip()
+            num_files = int(num_files) if num_files else None
+            hearing_impaired = tr.find("td", class_="a41") is not None
+
+            subtitles.add(SubsceneSubtitle(lang, hearing_impaired, page_link,
+                                           None, desc, None, release_name,
+                                           None, imdb_id, num_files, year))
+
+        return subtitles
+
+    def list_subtitles(self, video, languages):
+        if isinstance(video, Episode):
+            query = "%s - %s Season" % (video.series, to_ordinal[video.season])
+        else:
+            query = video.title
+        return self.query(query, video.year, languages)
+
+    def download_subtitle(self, subtitle):
+        logger.info("Downloading subtitle %r", str(subtitle.id))
+
+        url = self._baseurl + subtitle.page_link
+        with self._session.get(url) as response:
+            response.raise_for_status()
+            html = response.text
+        link = soupify(html).find("a", {"id": "downloadButton"})
+
+        subtitle.zip_link = link["href"]
+
+        logger.info("Download link: %r", subtitle.zip_link)
+        url = self._baseurl + subtitle.zip_link
+        with self._session.get(url) as response:
+            response.raise_for_status()
+            buffer = response.content
+
+        logger.info("Extracting downloaded subtitle")
+        with ZipFile(BytesIO(buffer)) as zipfile:
+            filenames = zipfile.namelist()
+            if subtitle.num_files is not None:
+                assert len(filenames) == subtitle.num_files
+            else:
+                subtitle.num_files = len(filenames)
+
+            if subtitle.num_files == 1:
+                filename = filenames[0]
+            else:
+                # TODO: support multi-file subtitles using guess matching
+                raise NotImplementedError()
```

Comment:
Is it common to have multiple files in an archive?

Reply:
Yeah it is. Mostly used with packs, though.

Reply:
So, in the end it has the same issue as legendastv provider.
Is it possible to know all the releases that an archive has prior to its download?

Reply:
There is a comment section on each subtitle's page giving descriptions for matching video names; although rarely reliable source of information. For example, there are complete packs of subtitles for a whole season but give absolutely no useful guess; sometimes just number of files.
[example](https://subscene.com/subtitles/breaking-bad-fourth-season-2011/english/797795)

So no.